core/src/main/java/tc/oc/pgm/projectile/ProjectileDefinition.java

```diff
@@ -50,6 +56,31 @@ public ProjectileDefinition(
     this.throwable = throwable;
     this.precise = precise;
     this.blockMaterial = blockMaterial;
+    this.scale = scale;
+    this.solidBlockCollision = solidBlockCollision;
+    this.maxTravelTime = maxTravelTime;
+  }
+
+  public static sealed class ProjectileEntity {
+    public static final class RealEntity extends ProjectileEntity {
+      public final Class<? extends Entity> entityType;
+
+      public RealEntity(Class<? extends Entity> entityType) {
+        this.entityType = entityType;
+      }
+    }
+
+    public static final class AbstractEntity extends ProjectileEntity {
+      public final AbstractEntityType entityType;
+
+      public AbstractEntity(AbstractEntityType entityType) {
+        this.entityType = entityType;
+      }
+    }
```

Comment:
You can probably use a record for these:

```suggestion
    record RealEntity(Class<? extends Entity> entityType) implements ProjectileEntity {}
    record CustomEntity(CustomEntityType type) implements ProjectileEntity {}
```